projects/core/src/cart/store/effects/wish-list.effect.ts

```diff
@@ -45,9 +54,76 @@ export class WishListEffects {
     })
   );
 
+  @Effect()
+  loadWishList$: Observable<
+    | CartActions.LoadWisthListSuccess
+    | CartActions.CreateWishList
+    | CartActions.LoadCartFail
+  > = this.actions$.pipe(
+    ofType(CartActions.LOAD_WISH_LIST),
+    map((action: CartActions.LoadWisthList) => action.payload),
+    concatMap(userId => {
+      return this.cartConnector.loadAll(userId).pipe(
+        switchMap(carts => {
+          if (carts) {
+            const wishList = carts.find(cart => cart.name === 'wishlist');
+            if (Boolean(wishList)) {
+              return [
+                new CartActions.LoadWisthListSuccess({
+                  cart: wishList,
+                  userId,
+                }),
+              ];
+            } else {
+              return [
+                new CartActions.CreateWishList({ userId, name: 'wishlist' }),
+              ];
+            }
+          }
+        }),
+        catchError(error => from([new CartActions.LoadCartFail(error)]))
+      );
+    })
+  );
+
+  @Effect()
+  resetWishList$: Observable<
+    CartActions.ResetWishListDetails
+  > = this.actions$.pipe(
+    ofType(
+      SiteContextActions.LANGUAGE_CHANGE,
+      SiteContextActions.CURRENCY_CHANGE
+    ),
+    switchMap(() => [new CartActions.ResetWishListDetails()])
+  );
+
+  @Effect()
+  getWishList$: Observable<
+    CartActions.LoadWisthListSuccess | CartActions.LoadCartFail
+  > = this.actions$.pipe(
+    ofType(CartActions.RESET_WISH_LIST_DETAILS),
+    withLatestFrom(
+      this.authService.getOccUserId(),
+      this.wishListService.getWishListId()
+    ),
+    switchMap(([, userId, wishListId]) => {
+      if (Boolean(wishListId)) {
+        return this.cartConnector.load(userId, wishListId).pipe(
+          switchMap(wishList => [
+            new CartActions.LoadWisthListSuccess({ cart: wishList, userId }),
+          ]),
+          catchError(error => from([new CartActions.LoadCartFail(error)]))
+        );
+      }
+      return EMPTY;
+    })
+  );
+
   constructor(
     private actions$: Actions,
     private cartConnector: CartConnector,
-    private saveCartConnector: SaveCartConnector
+    private saveCartConnector: SaveCartConnector,
+    private wishListService: WishListService,
```

Comment:
I am not sure whether it's better to have `private store$: Store<>` instead of `wishListService`. I think the facade layer is above store. Now, it makes the effect depend on the facade.